ksqldb-engine/src/main/java/io/confluent/ksql/query/QueryExecutor.java

```diff
@@ -122,7 +137,9 @@
       final FunctionRegistry functionRegistry,
       final KafkaStreamsBuilder kafkaStreamsBuilder,
       final StreamsBuilder streamsBuilder,
-      final MaterializationProviderBuilderFactory materializationProviderBuilderFactory
+      final MaterializationProviderBuilderFactory materializationProviderBuilderFactory,
```

Comment:
It seems the only reason we are passing in this streamsBuilder in the line above is for testing (the only  non-testing caller always pass in a `new TopologyStreamsBuilder`). And now with the new call path this object is not always used and hence is a bit awkward to still maintain it as a class field.

How about not maintaining it at all, and pass in one at the `buildPersistentQuery` and let the caller `QueryRegistryImpl` to determine which StreamsBuilder to pass in based on the `ksqlConfig`. We can even just have two the two projected `build(Not)SharedPersistentQuery` as public and let the `QueryRegistryImpl` to determine which one to call.

Reply:
Seems not addressed yet?

Reply:
Ah I miss-understood your suggestion. I pretty much did the second half but did not actually remove the global var. However I just updated it so it should be ready for another pass. 

`StreamsBuilder` is also used in the transient query creation but you are right that only is testing is it inputed, otherwise it is just the default. As far as I can tell it was only used in one way too. 

Eventually I would like to have the named wrapper to take a normal topology as well. Then we only need one type and we can make it a global var again and it will be simple again. But that can wait.